test/unit_tests/download/test_accounts.py

```diff
@@ -21,3 +25,42 @@ def test_add_parse_get_identity(monkeypatch, tmpdir):
 
     assert user_name == "user_name"
     assert password == "abcd"
+
+
+def test_parse_identity_failure(monkeypatch, tmpdir):
+    """
+    This test reads the identity file from the ``test_data`` folder and
+    tries to authenticate with a wrong password. This test asserts that
+    that an exception is thrown.
+    """
+    test_data = Path(PurePath(__file__).parent / "test_data" / "identities.json")
+    import pansat.download.accounts as accs
+
+    monkeypatch.setattr("os.environ", {})
+    monkeypatch.setattr("pansat.download.accounts._IDENTITY_FILE", test_data)
+    monkeypatch.setattr("pansat.download.accounts._PANSAT_SECRET", None)
+    monkeypatch.setattr("getpass.getpass", lambda: "abcd")
+
+    with pytest.raises(Exception):
+        accs.parse_identity_file()
+        accs.authenticate()
+
+
+HAS_PANSAT_PASSWORD = "PANSAT_PASSWORD" in os.environ
+
+
+@pytest.mark.skipif(not HAS_PANSAT_PASSWORD, reason="Pansat password not set.")
+def test_parse_identity(monkeypatch, tmpdir):
```

Comment:
How is this test supposed to work though? If I want to run it locally (and the environment variable will be have been set by the get_password function), then the environment variable will be my own pansat-password. Then every user would need to set up their own test_data/identities.json with their pansat password, right?
Also, the test-password should be set in the github workflow, so that the test can run there.

Reply:
Ok, just found the env variable under the secrets, so it should work in github. Regarding locally: is that why you don't set the environment variable in the code?

Reply:
Storing passwords in plain text on GitHub is a bad idea. So all this struggly is basically simply to avoid that.  For local tests you will have to set the environment variable manually but nothing stops you from doing this in your .bashprofile or .bashrc.